tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7488,11 +7488,31 @@ SpirvEmitter::processIntrinsicCallExpr(const CallExpr *callExpr) {
                                callExpr->getExprLoc());
       }
     }
-    spvBuilder.createRayTracingOpsNV(
-        hlslOpcode == hlsl::IntrinsicOp ::IOP_AcceptHitAndEndSearch
-            ? spv::Op::OpTerminateRayNV
-            : spv::Op::OpIgnoreIntersectionNV,
-        QualType(), {}, srcLoc);
+    bool nvRayTracing =
+        featureManager.isExtensionEnabled(Extension::NV_ray_tracing);
+
+    if (nvRayTracing) {
+      spvBuilder.createRayTracingOpsNV(
+          hlslOpcode == hlsl::IntrinsicOp::IOP_AcceptHitAndEndSearch
+              ? spv::Op::OpTerminateRayNV
+              : spv::Op::OpIgnoreIntersectionNV,
+          QualType(), {}, srcLoc);
```

Comment:
Based on the comment in the line 7501, both `OpTerminateRayKHR` and `OpIgnoreIntersectionKHR` are termination instructions. Is it ok that we do not create a new block here?

Reply:
Do you mean `OpTerminateRayKHR` and `OpIgnoreIntersectionNV` ?

Reply:
The KHR extension clearly requires that `OpTerminateKHR` and `OpIgnoreIntersectionKHR` be the last instruction in the basic block (and hence added a new block on line 7512 for them).
Reference: https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/KHR/SPV_KHR_ray_tracing.asciidoc

However, The NV equivalents don't seem to have this requirement in the spec.
Reference:  https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/NV/SPV_NV_ray_tracing.asciidoc

Reply:
@jaebaek I think this is why these ops (contrary to others) have a different ID than their NV counterpart, because they are semantically different.

Reply:
Makes sense. Thank you for doublechecking it.